e2e/test/E2ETestBase.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    /// <summary>
+    /// This class creates an instance of the logger for each test and logs the test result along with other useful information.
+    /// </summary>
+    public class E2ETestBase
+    {
+        protected TestLogging Logger { get; set; }
+        protected Stopwatch StopWatch { get; set; }
+        public TestContext TestContext { get; set; }
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            StopWatch = new Stopwatch();
+            StopWatch.Start();
+            Logger = TestLogging.GetInstance(TestContext);
+            Logger.WriteLine($"Starting test - {TestContext.TestName}", SeverityLevel.Information);
+        }
+
+        [TestCleanup]
+        public void TestCleanup()
+        {
+            string testStatus = TestContext.CurrentTestOutcome.ToString();
+            StopWatch.Stop();
+            double elapsed = StopWatch.Elapsed.TotalSeconds;
+            var properties = new Dictionary<string, string>
+            {
+                { LoggingPropertyNames.TestRunTime, elapsed.ToString() }
```

Comment:
Suggest putting the test status as a field in properties too, that way it can be easily queried for.

Reply:
TestStatus is a customDimension too. Is that what you are asking about?